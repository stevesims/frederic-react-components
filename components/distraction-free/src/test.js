import React from 'react';
import { mount, shallow } from 'enzyme';
import { GREY_1 } from 'govuk-colours';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

import DistractionFree from '.';

expect.extend(createMatchers(emotion));

describe('DistractionFree', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<DistractionFree />);
    expect(wrapper.exists()).toBe(true);
  });
  
  it('renders children text', () => {
    const wrapper = shallow(<DistractionFree>Text</DistractionFree>);
    expect(wrapper.find('Body').childAt(0).text()).toBe('Text');
  });
  
  it('renders optional arrow title', () => {
    const wrapper = shallow(<DistractionFree arrowLeftTitle="Title">Text</DistractionFree>);
    expect(wrapper.find('ArrowLeft').childAt(0).text()).toBe('Title');
  });
  
  it('renders function passed when clicked', () => {
    const clickHandler = jest.fn();
    const wrapper = shallow(<DistractionFree onClick={clickHandler}>Text</DistractionFree>);
    wrapper.find('Button').simulate('click');
    expect(clickHandler).toHaveBeenCalledTimes(1);
  });

  it('renders style rules', () => {
    const wrapper = shallow(<DistractionFree />);
    expect(wrapper.find('Background')).toHaveStyleRule('background-color', GREY_1);
  });
  
  it('matches snapshot', () => {
    const clickHandler = jest.fn();
    const wrapper = mount(<DistractionFree arrowLeftTitle="Title" onClick={clickHandler}>Text</DistractionFree>);
    expect(wrapper).toMatchSnapshot();
  });
});

