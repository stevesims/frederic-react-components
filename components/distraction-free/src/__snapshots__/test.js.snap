// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistractionFree matches snapshot 1`] = `
.emotion-10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #6f777b;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  left: 0;
  padding: 0 32px 0 90px;
}

.emotion-6 {
  max-width: 1158px;
  padding: 20px 0;
  width: 100%;
}

.emotion-6 > span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-4 {
  background-color: transparent;
  border: 0;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  color: #ffffff;
}

.emotion-0 {
  font-family: "nta",Arial,sans-serif;
  font-size: 14px;
  line-height: 1.1428571429;
  margin-left: 6px;
}

@media only screen and (min-width:641px) {
  .emotion-0 {
    font-size: 19px;
    line-height: 1.25;
  }
}

.emotion-8 {
  background-color: #ffffff;
  font-family: "nta",Arial,sans-serif;
  height: 100%;
  max-width: 1158px;
  min-height: 640px;
  width: 100%;
}

<DistractionFree
  arrowLeftTitle="Title"
  onClick={[MockFunction]}
>
  <Background>
    <div
      className="emotion-10 emotion-11"
    >
      <Header>
        <header
          className="emotion-6 emotion-7"
        >
          <Button
            onClick={[MockFunction]}
          >
            <button
              className="emotion-4 emotion-5"
              onClick={[MockFunction]}
            >
              <ArrowLeft
                color="#ffffff"
                fill="#ffffff"
                width={28}
              >
                <Wrapper
                  color="#ffffff"
                >
                  <span
                    className="emotion-2 emotion-3"
                    color="#ffffff"
                  >
                    <svg
                      fill="#ffffff"
                      height="28px"
                      viewBox="0 0 512 512"
                      width="28px"
                    >
                      <title>
                        Title
                      </title>
                      <path
                        d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zm28.9-143.6L209.4 288H392c13.3 0 24-10.7 24-24v-16c0-13.3-10.7-24-24-24H209.4l75.5-72.4c9.7-9.3 9.9-24.8.4-34.3l-11-10.9c-9.4-9.4-24.6-9.4-33.9 0L107.7 239c-9.4 9.4-9.4 24.6 0 33.9l132.7 132.7c9.4 9.4 24.6 9.4 33.9 0l11-10.9c9.5-9.5 9.3-25-.4-34.3z"
                      />
                    </svg>
                    <Title>
                      <span
                        className="emotion-0 emotion-1"
                      >
                        Title
                      </span>
                    </Title>
                  </span>
                </Wrapper>
              </ArrowLeft>
            </button>
          </Button>
        </header>
      </Header>
      <Body>
        <article
          className="emotion-8 emotion-9"
        >
          Text
        </article>
      </Body>
    </div>
  </Background>
</DistractionFree>
`;
